modules/ppcp-vaulting/src/PaymentTokenChecker.php

```diff
@@ -143,6 +143,8 @@ public function check_and_update( int $order_id, int $customer_id, string $inten
 
 		try {
 			$subscription_behavior_when_fails = $this->settings->get( 'subscription_behavior_when_vault_fails' );
+			$wc_order->update_meta_data( self::VAULTING_FAILED_META_KEY, $subscription_behavior_when_fails );
+			$wc_order->save_meta_data();
 		} catch ( NotFoundException $exception ) {
 			return;
```

Comment:
should we just return when the value is missing? I think it can happen when updated the plugin but not re-saved the settings. Probably should do the old behavior in this case.

Reply:
Good catch! I´ve just added it, thanks :)